job-server/src/main/scala/spark/jobserver/JobServer.scala

```diff
@@ -90,6 +97,36 @@ object JobServer {
     }
   }
 
+  private def maskCredentials(lines: String, credentialRegex: Regex): String = {
+    lines
+      .split("\n")
+      .toSeq
+      .map {
+        line => line.split(":") match {
+          // if key matches credential keys pattern and value is not empty, mask credentials
+          case Array(key, value) if (credentialRegex.findFirstIn(key).nonEmpty
+            && EMPTY_VALUE_PATTERN.findFirstIn(value.stripMargin).isEmpty) =>
+            Array(key, CREDENTIAL_MASK).mkString(":")
+          case _ => line
+      }
+    }.mkString("\n")
+  }
+
+  private def credentialRegex(config: Config): Regex = {
+    // Use default credential keys if spark.ui.confidentialKeys is not set
+    val credentialKeys = try {
+      config.getString("spark.ui.confidentialKeys")
```

Comment:
NIT: do we want to have this defined in config file? (to make modifications easier for users)

Reply:
yes, it's from spark-defaults.conf